lib/makara/cache.rb

```diff
@@ -16,10 +16,13 @@ def store=(store)
       end
 
       def read(key)
-        store.try(:read, key)
+        store.try(:read, key).tap do |value|
```

Comment:
I only use `tap` when there is data that I want to instrument that is not directly accessible as a one-liner. For instance, with `#write` all of the values are there from the method signature. With `#read`, the key is there from the method, signature, but value is obtained from the memory store.

The alternative would be:

```ruby
value = store.try(:read, key)
Notifications.notify!('Cache:read', key, value)
value
```